tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/XMLQueryExpressionTest.java

```diff
@@ -48,8 +48,8 @@ public void setup() {
     public void testNegativeQueryExprForXML() {
         int index = 0;
         validateError(negativeResult, index++, "incompatible types: expected " +
-                        "'xml<((xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text) & readonly)> & readonly'," +
-                        " found 'xml'", 21, 16);
+                "'xml<((xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text) & readonly)> & readonly'," +
+                " found '(xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text)'", 21, 16);
```

Comment:
isn't previous error message make more sense to the user?

Reply:
For query expression it returns one of `xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text`. Isn't this error message giving correct information to the user